smack-extensions/src/main/java/org/jivesoftware/smackx/usertune/element/UserTuneElement.java

```diff
@@ -0,0 +1,254 @@
+         * Length is an optional element in UserTuneElement.
+         * @param length
+         * @return Builder
+         */
+        public Builder setLength(int length) {
+            if (length > 0) {
+                this.length = length;
+                return this;
+            }
+        	else {
+        		throw new IllegalArgumentException("Length shoud be positive. Incase of live streaming, length is expected to be null");
+        	}
+        }
+
+        /**
+         * Rating is an optional element in UserTuneElement.
+         * The rating to be given is suppossed to be out of 10.
+         * <br>
+         * Method will throw an Exception if rating is less than 0 or greater than 10.
+         * @param rating
+         * @return Builder
+         */
+        public Builder setRating(int rating) {
+            if (rating <= 10 && rating >= 0) {
+                this.rating = rating;
+            } else {
+                throw new IllegalArgumentException("Rating tried to set was out of bounds");
+            }
+            return this;
+        }
+
+        /**
+         * Source is an optional element in UserTuneElement.
+         * @param source
+         * @return Builder
+         */
+        public Builder setSource(String source) {
+            this.source = source;
+            return this;
+        }
+
+        /**
+         * Title is a compulsory element in UserTuneElement.
+         * @param title
+         * @return Builder
+         */
+        public Builder setTitle(String title) {
+            this.title = title;
+            return this;
+        }
+
+        /**
+         * Track is an optional element in UserTuneElement.
+         * @param track
+         * @return Builder
+         */
+        public Builder setTrack(int track) {
+        	if (track > 0) {
+                this.track = track;
+                return this;
+            }
+        	else {
+                throw new IllegalArgumentException("Track number should be positive.");
+            }
+        }
+
+        public Builder setUri(URI uri) {
+            this.uri = uri;
+            return this;
+        }
+
+        /**
+         * This method is called to build a UserTuneElement.
+         * @throws IllegalArgumentException
+         * @return UserTuneElement
+         */
+        public UserTuneElement build() {
+            if (title != null) {
+                return new UserTuneElement(this);
+            }
+            else {
+                throw new IllegalArgumentException("UserTune should atleast have 'title'");
+            }
+        }
+
+        /**
+         * This method is called to build a null UserTuneElement while disabling publish.
+         * @throws IllegalArgumentException
+         * @return UserTuneElement
+         */
+        public UserTuneElement buildEmptyUserTune() {
+            if (title == null || source == null || uri == null || artist == null || length == null || rating == null || track == null) {
```

Comment:
I would simply return a new UserTuneElement without any values. Maybe add another private constructor for this.